### PR TITLE
[AppService] Fix #16681: az webapp config ssl import: fixing bug that causes failures on national clouds 

### LINKED ISSUE #16681: [Regression]:Azure Government | az webapp config ssl import returns 'Microsoft.CertificateRegistration' is invalid
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
Using 2.18.0 of the CLI in Azure Government yields the error below when trying to add a keyvault cert to an app service. Downgrading to 2.17.0 works successfully.

error:
_ResourceNotFoundError: InvalidResourceNamespace: The resource namespace 'Microsoft.CertificateRegistration' is invalid._

**To Reproduce**
1. Install Azure CLI version 2.18.0

2. run the following command
az webapp config ssl import --resource-group rg--name appname--key-vault kvname --key-vault-certificate-name certname

**Expected behavior**

**Environment summary**
jeremy@Azure:~$ az --version
azure-cli                         2.18.0

core                              2.18.0
telemetry                          1.0.6

Extensions:
ai-examples                        0.2.5
azure-cli-ml                      1.20.0

Python location '/opt/az/bin/python3'
Extensions directory '/home/jeremy/.azure/cliextensions'
Extensions system directory '/opt/az/lib/python3.6/site-packages/azure-cli-extensions'

**Additional context**
jeremy@Azure:~$ az webapp config ssl import --resource-group rg --name appname --key-vault kvname --key-vault-certificate-name testcert --debug

msrest.async_paging: Paging async iterator protocol is not available for AppServiceCertificateOrderPaged
cli.azure.cli.core.adal_authentication: AdalAuthentication.signed_session invoked by Track 1 SDK
cli.azure.cli.core._profile: Retrieving token from ADAL for resource 'https://management.core.usgovcloudapi.net/'
urllib3.connectionpool: Starting new HTTP connection (1): localhost:50342
urllib3.connectionpool: http://localhost:50342 "POST /oauth2/token HTTP/1.1" 200 1886
msrestazure.azure_active_directory: MSI: Retrieving a token from http://localhost:50342/oauth2/token, with payload {'resource': 'https://management.core.usgovcloudapi.net/'}
urllib3.connectionpool: Starting new HTTP connection (1): localhost:50342
urllib3.connectionpool: http://localhost:50342 "POST /oauth2/token HTTP/1.1" 200 1886
msrestazure.azure_active_directory: MSI: Retrieving a token from http://localhost:50342/oauth2/token, with payload {'resource': 'https://management.core.usgovcloudapi.net/'}
.........<redacted>
msrest.http_logger: Request URL: 'https://management.usgovcloudapi.net/subscriptions/xxx/providers/Microsoft.CertificateRegistration/certificateOrders?api-version=2019-08-01'
msrest.http_logger: Request method: 'GET'
msrest.http_logger: Request headers:
msrest.http_logger:     'Accept': 'application/json'
msrest.http_logger:     'accept-language': 'en-US'
msrest.http_logger:     'User-Agent': 'python/3.6.10 (Linux-4.15.0-1103-azure-x86_64-with-debian-10.2) msrest/0.6.18 cloud-shell/1.0 msrest_azure/0.6.3 azure-mgmt-web/0.48.0 Azure-SDK-For-Python AZURECLI/2.18.0 (DEB)'
msrest.http_logger: Request body:
msrest.http_logger: None
msrest.universal_http: Configuring redirects: allow=True, max=30
msrest.universal_http: Configuring request: timeout=100, verify=True, cert=None
msrest.universal_http: Configuring proxies: ''
msrest.universal_http: Evaluate proxies against ENV settings: True
urllib3.connectionpool: https://management.usgovcloudapi.net:443 "GET /subscriptions/xxxx/providers/Microsoft.CertificateRegistration/certificateOrders?api-version=2019-08-01 HTTP/1.1" 404 128
msrest.http_logger: Response status: 404
msrest.http_logger: Response content:
msrest.http_logger: {"error":{"code":"InvalidResourceNamespace","message":"The resource namespace 'Microsoft.CertificateRegistration' is invalid."}}
msrest.exceptions: (InvalidResourceNamespace) The resource namespace 'Microsoft.CertificateRegistration' is invalid.
cli.azure.cli.core.util: azure.cli.core.util.handle_exception is called with an exception:
cli.azure.cli.core.util: Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 659, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 722, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 714, in _run_job
    return cmd_copy.exception_handler(ex)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/commands.py", line 37, in _ex_handler
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 693, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 330, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 786, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 2447, in import_ssl_cert
    for asc in ascs:
  File "/opt/az/lib/python3.6/site-packages/msrest/paging.py", line 143, in __next__
    self.advance_page()
  File "/opt/az/lib/python3.6/site-packages/msrest/paging.py", line 129, in advance_page
    self._response = self._get_next(self.next_link)
  File "/opt/az/lib/python3.6/site-packages/azure/mgmt/web/v2019_08_01/operations/_app_service_certificate_orders_operations.py", line 97, in internal_paging
    raise models.DefaultErrorResponseException(self._deserialize, response)
azure.mgmt.web.v2019_08_01.models._models_py3.DefaultErrorResponseException: (InvalidResourceNamespace) The resource namespace 'Microsoft.CertificateRegistration' is invalid.

cli.azure.cli.core.azclierror: ResourceNotFoundError: InvalidResourceNamespace: The resource namespace 'Microsoft.CertificateRegistration' is invalid.
cli.azure.cli.core.azclierror: ResourceNotFoundError: InvalidResourceNamespace: The resource namespace 'Microsoft.CertificateRegistration' is invalid.
cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x7fecbc379158>]
az_command_data_logger: exit code: 3
cli.__main__: Command ran in 1.060 seconds (init: 0.116, invoke: 0.944)
telemetry.save: Save telemetry record of length 3230 in cache
telemetry.check: Negative: The /home/jeremy/.azure/telemetry.txt was modified at 2021-01-27 01:43:50.280020, which in less than 600.000000 s
jeremy@Azure:~$

### PR DESCRIPTION
**Pull request by singlessis**
**Description**

With the last a change, a bug was introduced that attempted to use the CertificateRegistration RP in national clouds, which causes a failure.

**Testing Guide**

az webapp config ssl import --resource-group rg --name app --key-vault kv --key-vault-certificate cert

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -2436,18 +2436,22 @@ def import_ssl_cert(cmd, resource_group_name, name, key_vault, key_vault_certifi
     kv_resource_group_name = kv_id_parts['resource_group']
     kv_subscription = kv_id_parts['subscription']
 
-    if kv_subscription.lower() != client.config.subscription_id.lower():
-        diff_subscription_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_APPSERVICE,
-                                                           subscription_id=kv_subscription)
-        ascs = diff_subscription_client.app_service_certificate_orders.list()
+    cloud_type = cmd.cli_ctx.cloud.name
+    if cloud_type.lower() == "AzureCloud".lower():
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:2440]: Is Azure Cloud public? if so this logic is incorrect. Note: the change expected is
client.app_service_certificate_orders -> is supported on public only so making this call should be avoided on all non-public clouds.

**Review by panchagnula — CHANGES_REQUESTED** [review]
Please confirm the logic for checking the environment is right

**Comment by singlessis** [comment]
@qwordy @Juliehzl I am trying to write a test that runs in a national cloud, to track this scenario going forward, but it seems like the way I managed to get it working locally is not ideal, it might contain sensitive information. Is there a better way to run a test on a government cloud? thanks!

**Comment by panchagnula** [comment]
> @qwordy @Juliehzl I am trying to write a test that runs in a national cloud, to track this scenario going forward, but it seems like the way I managed to get it working locally is not ideal, it might contain sensitive information. Is there a better way to run a test on a government cloud? thanks!

@qwordy  mainly is there a way we can run specific tests or all test on non public clouds. That way Sofi doesn't need to add a specific test for S-clouds?

**Review by panchagnula — APPROVED** [review]
Lets get the build to pass otherwise looks good to me.

**Comment by singlessis** [comment]
> Lets get the build to pass otherwise looks good to me.

The build is failing because of the test I tried adding, I will remove it, and hopefully we can run the tests in national cloud

**Comment by qwordy** [comment]
@singlesis Login to the cloud you want to test. Then run test lively. Replay test should be able to pass as long as live test can pass. If code in test case has sensitive information, I recommend deleting the test case. But, remember to make sure it can work in the expected cloud. If recording file has sensitive information, you can mark the test case as `live_only()`, then delete the recording file.

**Review by qwordy — APPROVED** [review]

**Comment by singlessis** [comment]
@Juliehzl could you take a look?

**Pull request merged**